src/main/java/com/uber/cadence/converter/JsonDataConverter.java

```diff
@@ -208,6 +209,35 @@ public T read(JsonReader in) throws IOException {
           }
         };
       }
+      if (Class.class.isAssignableFrom(typeToken.getRawType())) {
+        return new TypeAdapter<T>() {
+          @Override
+          public void write(JsonWriter out, T value) throws IOException {
+            out.beginObject();
+            String className = value.getClass().getName();
```

Comment:
This should be ((Class) value).getName();
value.getClass().getName() is always java.lang.Class
Let's add a DataConverter to test this directly?

Reply:
Good catch!